discord/discord_client.py

```diff
@@ -0,0 +1,72 @@
+from django.conf import settings
+from django.template.defaultfilters import slugify
+
+from disco.api.client import APIClient
+from disco.types.channel import ChannelType
+
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+class InstantiationError(Exception):
+    pass
+
+
+class DiscordClient:
+    """
+    Singleton class wrapping Discord API functionality.
+    """
+
+    __instance = None
+
+    @classmethod
+    def get_instance(cls):
+        """Return singleton client object."""
+        if not cls.__instance and cls._has_valid_settings():
+            cls.__instance = cls(settings.DISCORD_API_TOKEN, settings.DISCORD_GUILD_ID)
+        return cls.__instance
+
+    @staticmethod
+    def _has_valid_settings():
+        return (
+            settings.DISCORD_API_TOKEN is not None
+            and settings.DISCORD_GUILD_ID is not None
+        )
+
+    def __init__(self, api_token, guild_id):
+        if self.__class__.__instance:
+            raise InstantiationError("Singleton object already created.")
+        self._client = APIClient(api_token)
+        self._guild_id = guild_id
+
+    def create_channel(self, name, chan_type=ChannelType.GUILD_TEXT, parent_name=None):
```

Comment:
Can we document what kinds of names are supported? (Emojis? special chars?)

What happens when a channel with the same name exists?

Reply:
Yep, can do. Looks like Discord supports emojis and special characters, but converts spaces to hyphens, so we can probably remove the slugify logic and directly request the name of the puzzle. WDYT?

Reply:
For existing channels with same names, we currently just create a new channel with the same name. To ensure channel name uniqueness, we can probably use the `get_or_create_channel()` method instead of `create_channel()` and maybe add a `prune_redundant_channels()` somewhere.